Problematic exception specification for what()

https://github.com/awdeorio/csvstream/blob/13f65a372f3f1776665474c92adf1350505d86d3/csvstream.h#L26

[Report from student](https://piazza.com/class/jpyjpprthr26wf?cid=4828#):
```
./csvstream.h:25:7: error: exception specification of overriding
      function is more lax than base version
class csvstream_exception : public std::exception {
```

Not sure what system or compiler version. Will update if I found out.

Possible fix:
```
  const char * what () const noexcept { 
```

--- a/csvstream.h
+++ b/csvstream.h
@@ -23,7 +23,7 @@
 // A custom exception type
 class csvstream_exception : public std::exception {
 public:
-  const char * what () const throw () {
+  const char * what () const noexcept {
     return msg.c_str();
   }
   const std::string msg;

